src/fastoad/notebooks/03_custom_modules/beam_problem.ipynb

```diff
@@ -0,0 +1,378 @@
+    "* The height of the beam for a given moment of inertia and width. \n",
+    "* The linear weight **w** from material density and section dimensions. \n",
+    "* The necessary moment of inertia to reach a given displacement. \n",
+    "* The necessary moment of inertia not to overcome the material yield stress. \n",
+    "\n",
+    "As you may notice, each module is registered using the decorator `@oad.RegisterOpenMDAOSystem` and a unique id. \n",
+    "\n",
+    "An example is provided for the geometry module that computes the height of the beam for a given moment of inertia and a given width: \n",
+    "\n",
+    "> ``` python\n",
+    "> import numpy as np\n",
+    "> import openmdao.api as om\n",
+    ">\n",
+    "> import fastoad.api as oad\n",
+    ">\n",
+    ">\n",
+    "> @oad.RegisterOpenMDAOSystem(\"tutorial.beam_problem.geometry\")\n",
+    "> class RectangularSection(om.ExplicitComponent):\n",
+    ">     def setup(self):\n",
+    ">\n",
+    ">         self.add_input(\"data:geometry:l\", val=np.nan, units=\"m\")\n",
+    ">         self.add_input(\"data:geometry:Ixx\", val=np.nan, units=\"m ** 4\")\n",
+    ">         self.add_output(\"data:geometry:h\", val=0.01, units=\"m\")\n",
+    ">\n",
+    ">         self.declare_partials(\"*\", \"*\", method=\"fd\")\n",
+    ">\n",
+    ">     def compute(self, inputs, outputs):\n",
+    ">         l = inputs[\"data:geometry:l\"]\n",
+    ">         I_xx = inputs[\"data:geometry:Ixx\"]\n",
+    ">\n",
+    ">         outputs[\"data:geometry:h\"] = (12 * I_xx / l) ** (1 / 3)\n",
+    "> ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Configuration file\n",
+    "Once the modules are created, the problem can be defined through the [configuration file](./data/beam_problem.yml): \n",
+    "* The folder(s) containing your custom modules is (are) indicated within the `module_folders` section.\n",
+    "* Your problem is defined within the `model` section. \n",
+    "* The order of the modules in the `model` section will drive the order of execution. If at least one output variable of a module is an input of a previous one, a loop occurs and and `nonlinear_solver` and `linear_solver` must be defined. Here non linear Block Gauss-Seidel and linear Direct solvers are used. \n",
+    "\n",
+    "The configuration file for the first problem is shown below: "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> ```yaml\n",
+    "> title: Tutorial Beam Problem\n",
+    ">\n",
+    "> # List of folder paths where user added custom registered OpenMDAO components\n",
+    "> module_folders:\n",
+    ">     - ../modules\n",
+    ">\n",
+    "> # Input and output files\n",
```

Comment:
For sake of clarity, I wonder if it would be useful to split the display of the configuration file:
- The first part would show the file up to the `module_folders` part. At this point, you may show the `list_modules` command.
- then the other part of the file could be displayed and I think the following text `"Then, as you can notice..."` would flow more naturally.

What do you think?